lib/ansible/module_utils/yumdnf.py

```diff
@@ -0,0 +1,122 @@
+        update_only=dict(required=False, default="no", type='bool'),
+        validate_certs=dict(type='bool', default=True),
+        # this should not be needed, but exists as a failsafe
+    ),
+    required_one_of=[['name', 'list']],
+    mutually_exclusive=[['name', 'list']],
+    supports_check_mode=True,
+)
+
+# 64k.  Number of bytes to read at a time when manually downloading pkgs via a url
+BUFSIZE = 65536
+
+
+class YumDnf(with_metaclass(ABCMeta, object)):
+    """
+    Abstract class that handles the population of instance variables that should
+    be identical between both YUM and DNF modules because of the feature parity
+    and shared argement spec
+    """
+
+    def __init__(self, module):
+
+        self.module = module
+
+        self.allow_downgrade = self.module.params['allow_downgrade']
+        self.autoremove = self.module.params['autoremove']
+        self.bugfix = self.module.params['bugfix']
+        self.conf_file = self.module.params['conf_file']
+        self.disable_excludes = self.module.params['disable_excludes']
+        self.disable_gpg_check = self.module.params['disable_gpg_check']
+        self.disable_plugin = self.module.params['disable_plugin']
+        self.disablerepo = self.module.params.get('disablerepo', [])
+        self.download_only = self.module.params['download_only']
+        self.enable_plugin = self.module.params['enable_plugin']
+        self.enablerepo = self.module.params.get('enablerepo', [])
+        self.exclude = self.module.params['exclude']
+        self.installroot = self.module.params['installroot']
+        self.install_repoquery = self.module.params['install_repoquery']
+        self.list = self.module.params['list']
+        self.names = [p.strip() for p in self.module.params['name']]
+        self.releasever = self.module.params['releasever']
+        self.security = self.module.params['security']
+        self.skip_broken = self.module.params['skip_broken']
+        self.state = self.module.params['state']
+        self.update_only = self.module.params['update_only']
+        self.update_cache = self.module.params['update_cache']
+        self.validate_certs = self.module.params['validate_certs']
+
+        # It's possible someone passed a comma separated string since it used
+        # to be a string type, so we should handle that
+        if self.enablerepo and len(self.enablerepo) == 1 and ',' in self.enablerepo:
+            self.enablerepo = self.module.params['enablerepo'].split(',')
+        if self.disablerepo and len(self.disablerepo) == 1 and ',' in self.disablerepo:
+            self.disablerepo = self.module.params['disablerepo'].split(',')
+        if self.exclude and len(self.exclude) == 1 and ',' in self.exclude:
+            self.exclude = self.module.params['exclude'].split(',')
+
+    @abstractmethod
+    def run(self):
+        raise NotImplementedError
+
+    def fetch_rpm_from_url(self, spec):
```

Comment:
Hopefully we'll move this into utils soon https://github.com/ansible/ansible/pull/19172